client/src/document/toolbar/flaws.tsx

```diff
@@ -92,7 +91,9 @@ function Flaws({ flaws }: { flaws: FlawCheck[] }) {
       {flaws.map((flaw) => {
         switch (flaw.name) {
           case "broken_links":
-            return <BrokenLinks key="broken_links" urls={flaw.flaws} />;
+            return (
+              <BrokenLinks key="broken_links" doc={doc} links={flaw.flaws} />
```

Comment:
Perhaps my brain is fried but I don't understand what you are saying. 
I do appreciate that there's something weird because when you're inside the `BrokenLinks` component you're pass two parameters: `doc` and `links`. And if you look closely you see that `doc.flaws.broken_links == links`. So it's like doing:
```
const user = {name: "peter"}
<BrokenLinks user={user} name={user.name}/>
```

So what should I do?

The `doc.flaws` is always wildly different for each kind. That's why I use `any` on line 82. 

So, I'd like to change it to `<BrokenLinks doc={doc}/>` then, within that component I can access the pieces I need which are: `doc.source.folder` and `doc.flaws.broken_links`.
But how do you even do that?! I tried:
```
const links: Link[] = doc.flaws.broken_links ? doc.flaws.broken_links : []
```
but that gives me:
```
Property 'broken_links' does not exist on type 'object'.  TS2339
```
:(
that's because, I guess, in `types.tsx` we only just set: `flaws: object;` which might not be good enough.

Reply:
Yeah `object` is a fairly restrictive type, I'd go for something that explicity spells out what you're expecting. Isn't it something like: `flaws: { broken_links: Link[] }`?

To make my point from before more explicit: The type definition of `<BrokenLinks>` says that you're expecting a `links` prop where each element of that list has a field named `nth` while in actuallity you're expecting that prop to not be there and are then patching it in. It doesn't cause a type error in how you use it (because you pass an `any`) in, but it does make for confusing code. So I'd recommend either changing the definition to something like `links: Omit<Link, 'nth'>` or I guess going with your idea of directly passing `doc` in